src/openscad.cc

```diff
@@ -448,46 +450,46 @@ int cmdline(const char *deps_output_file, const std::string &filename, const cha
 
 		fs::current_path(original_path);
 
-		if (stl_output_file) {
-			if (!checkAndExport(root_geom, 3, FileFormat::STL, stl_output_file)) {
+		if (curFormat == ExportFileFormat::STL) {
```

Comment:
Now that we have a mapping, we don't really need this huge if statement; we should be able to collect the similar calls and pass the format and filename arguments directly and thus simplify this logic (and duplication) a bit.